pandas/tests/indexes/test_base.py

```diff
@@ -2821,3 +2821,11 @@ def test_shape_of_invalid_index():
 
     idx = pd.Index([0, 1, 2, 3])
     assert idx[:, None].shape == (4, 1)
+
+
+def test_index_construction_respects_dtype():
+    index_list = [7606741985629028552, 17876870360202815256]
+    expected = np.asarray(index_list, dtype="uint64")
+    result = np.asarray(UInt64Index(index_list, dtype="uint64"), dtype="uint64")
```

Comment:
I think we can more directly test this by comparing two `UInt64Index`: one constructed from a list and one constructed from a `uint64` numpy array.  For example, the following fails on `master` and should be fixed here:
```python
In [1]: import numpy as np 
   ...: import pandas as pd 
   ...: import pandas.util.testing as tm 
   ...: pd.__version__
Out[1]: '0.26.0.dev0+842.g78c3c1640'

In [2]: index_list = [7606741985629028552, 17876870360202815256]

In [3]: result = pd.UInt64Index(index_list)

In [4]: expected = pd.UInt64Index(np.array(index_list, dtype="uint64"))

In [5]: tm.assert_index_equal(result, expected)
---------------------------------------------------------------------------
AssertionError: Index are different

Index values are different (100.0 %)
[left]:  UInt64Index([7606741985629028352, 17876870360202815488], dtype='uint64')
[right]: UInt64Index([7606741985629028552, 17876870360202815256], dtype='uint64')
```

Reply:
I just committed  what I think may be an even more direct test, prompted by other tests in the test class.